### PR TITLE
Update the Vector documentation

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: docs/content/en/tasks/monitor/vector/_index.md [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
 
-Since k8ssandra-operator v1.5.0, it is possible to scrape Cassandra metrics with the Vector agent and customize its configuration to add transformers and sinks.
+k8ssandra-operator provides a Vector instance that can be used to scrape logs and metrics. It's configuration is fully customizable and one can add transformers and sinks to provide the kind of solution user needs.
 
 More information about Vector can be found in the [official documentation](https://vector.dev/docs/).
 
 ## Enabling Vector agent
 
-To enable Vector agent, you need to add a `.spec.cassandra.telemetry` section in the `K8ssandraCluster` manifest:
+Vector agent is enabled by default for Cassandra pods. To enable Vector agent for all the other pods, you need to add a `.spec.cassandra.telemetry` section in the `K8ssandraCluster` manifest:
```
Comment (adejanovski) [inline docs/content/en/tasks/monitor/vector/_index.md:14]:
```suggestion
Vector agent is enabled by default for Cassandra pods. To enable Vector agent for Stargate and Reaper, you need to add a `.spec.stargate.telemetry` and a `.spec.reaper.telemetry` sections respectively in the `K8ssandraCluster` manifest with `.vector.enabled: true`:
```

The example below should probably not be using `.spec.cassandra`  but rather `.spec.stargate` for example.

File: docs/content/en/tasks/monitor/vector/_index.md [hunk 2]
```diff
@@ -107,7 +143,33 @@ The above configuration should display the scraped Cassandra metrics in json for
...
+                target = "stdout"
+                [sinks.console_output.encoding]
+                codec = "json"
+        scrapeInterval: 30s
+        resources:
+          requests:
+            cpu: 1000m
+            memory: 1Gi
+          limits:
+            memory: 2Gi
```
Comment (adejanovski) [inline docs/content/en/tasks/monitor/vector/_index.md:169]: These aren't taken into account, right?
Do we have a ticket to track making these values correctly mapped to the cassdc object (and what that mapping should be)?

**Review by adejanovski — CHANGES_REQUESTED** [review]

File: docs/content/en/tasks/monitor/vector/_index.md (hunk 2 above)
Comment (burmanm) [reply docs/content/en/tasks/monitor/vector/_index.md:169]: Well, they're read - but they apply them to the wrong place in the CassDc (they do it like the tests say.. but tests are wrong ;)). 

There's another PR, but no ticket.

Comment (burmanm) [reply docs/content/en/tasks/monitor/vector/_index.md:169]: No, there was a ticket.. https://github.com/k8ssandra/k8ssandra-operator/issues/940

**Review by adejanovski — APPROVED** [review]

**Pull request merged**